shared/src/main/scala/VNum.scala

```diff
@@ -46,14 +46,17 @@ object VNum:
     // Spits into real and imaginary parts
     // todo handle empty real part
     val parts = s.split("ı")
-    val real = parts(0)
-    val imag = parts.lift(1).getOrElse("0")
+    val real = parts.lift(0).getOrElse("0")
+    val imag = if s.endsWith("ı") then "1" else parts.lift(1).getOrElse("0")
 
-    val realNum = (if real.last == '.' then real + "5" else real).toInt
-    val imagNum = (if imag.last == '.' then imag + "5" else imag).toInt
+    // Apologies for not using val - I need to use the same variable twice
```

Comment:
You don't need to apologize for using vars lol, if they're the better way to do things, they should be used